Makefile

```diff
@@ -304,13 +317,39 @@ endif
 
 
 ### This target auto-generates a new grub.cfg file and uses grub to build a bootable ISO.
-### This target should be invoked when all of contents of `GRUB_ISOFILES` are ready to be packaged into an ISO.
+### This target should be invoked when all of contents of `BOOTLOADER_ISOFILES` are ready to be packaged into an ISO.
 grub:
-	@cargo run --release --manifest-path $(ROOT_DIR)/tools/grub_cfg_generation/Cargo.toml -- $(GRUB_ISOFILES)/modules/ -o $(GRUB_ISOFILES)/boot/grub/grub.cfg
-	@$(GRUB_MKRESCUE) -o $(iso) $(GRUB_ISOFILES)  2> /dev/null
-
-
-### This target copies all extra files into the `GRUB_ISOFILES` directory,
+	@if [ "$(GRUB_MKRESCUE)" = "not-found" ]; then \
+		echo "could not find grub-mkrescue!"; \
+		error "grub-mkrescue not found"; \
+	fi
```

Comment:
let's avoid using bash-specific features within `Make` as much as possible, as it hurts portability. See my other comment about this.